scripts/release/asset_publish.py

```diff
@@ -179,26 +178,45 @@ def validate_update_command_component(
 
     env = None
     # Check if component's env is registered
-    for version in [env_version, final_version]:
-        try:
-            if (env := mlclient.environments.get(name=env_name, version=version)) is not None:
-                break
-        except Exception as e:
-            logger.print(f"Fetching component env {env_name}:{version} from registry failed. Error: {e}")
+    registered_envs = get_registered_asset_versions(assets.AssetType.ENVIRONMENT.value, env_name, registry_name)
```

Comment:
container returns NextVersion, it is probably more robust to use that.

if you retrieve or attempt to register, you can also retrieve by label to resolve to the version to address your label support question

Reply:
Thanks @vizhur, as discussed offline I will move to using labels to check latest version, when support for the same is provided in CLI. Going with this approach for the time being